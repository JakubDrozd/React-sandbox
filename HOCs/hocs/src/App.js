import ClickCounter from "./ClickCounter";

function App() {
  return (
    <div>
      <ClickCounter></ClickCounter>
    </div>
  );
}

export default App;
