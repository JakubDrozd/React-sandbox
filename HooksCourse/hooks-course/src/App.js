import { useState, useReducer, useEffect } from "react";
import styled from "styled-components";
import axios from "axios";

//App component
const Base = styled.div`
  text-align: center;
`;

function App() {
  const [data, setData] = useState("");
  const [count, setCount] = useState(0);

  useEffect(() => {
    axios
      .get("https://jsonplaceholder.typicode.com/comments")
      .then((response) => {
        setData(response.data[0].email);
        console.log("API WAS CALLED");
      });
  }, []);

  return (
    <Base>
      <h1>{count}</h1>
      <button
        onClick={() => {
          setCount(count + 1);
        }}
      >
        Click Here
      </button>
      {<p>{data}</p>}
    </Base>
  );
}

export default App;
