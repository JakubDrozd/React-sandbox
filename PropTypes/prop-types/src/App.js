import { Greeting } from "./Greeting";
import styled from "styled-components";
import { useState } from "react";
import { Counter } from "./Counter";

const Button = styled.button`
  font-size: 1em;
  background: ${(props) => (props.dark ? "black" : "white")};
  color: ${(props) => (props.dark ? "white" : "black")};
  border: 1px solid ${(props) => (props.dark ? "gray" : "black")};
  padding: 0.25em 1em;
`;
function App() {
  const [buttonTheme, setButtonTheme] = useState(false);

  return (
    <div>
      <Greeting
        name="Jakub"
        repos={[
          { name: "project-shopping-car", index: 1 },
          { name: "react-sandbox", index: 2 },
          { name: "memory-card", index: 3 },
          { name: "typescript", index: 4 },
          { name: "cv-application", index: 5 },
          { name: "battleships", index: 6 },
          { name: "js-top", index: 7 },
        ]}
      ></Greeting>
      <Button dark={buttonTheme} onClick={() => setButtonTheme(!buttonTheme)}>
        Click me!
      </Button>
      <Counter></Counter>
    </div>
  );
}

export default App;
