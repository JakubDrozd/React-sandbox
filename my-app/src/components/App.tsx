//logic
import React from "react";

