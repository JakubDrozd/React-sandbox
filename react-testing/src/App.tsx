import { useEffect, useState } from "react";

function App() {
  let [counter, setCounter] = useState(0)
  return <button data-testid="button" onClick={()=> setCounter(counter + 1)}>{counter}</button>
}

export default App;
