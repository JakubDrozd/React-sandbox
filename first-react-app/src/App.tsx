import React from 'react';
import logo from './logo.svg';
import './App.css';


export function App() {
  return (
    <div className="App">
      <Welcome></Welcome>
      <Welcome></Welcome>
      <Welcome></Welcome>
    </div>
  );
}

const user = {
  firstName: "Jakub",
  lastName: "Drozd"
}

export function Welcome(){
  function formatName(user: {firstName: string, lastName: string}) {
    if(user){
      return `${user.firstName} ${user.lastName} it is ${new Date().toLocaleTimeString("en-GB")}`
    } 
    return <h1>Hello, anonymous user</h1>
  }
  
   function tick(){
    return <h1>Hello, {formatName(user)}</h1>
  }

  return tick()
  
}

