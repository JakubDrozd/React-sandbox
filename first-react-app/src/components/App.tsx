import '../../src/style/App.css';
import { Welcome } from './Welcome';

 
export function App(props: any) {
  return (
    <div className="App">
      <Welcome firstName="Jakub" lastName="Drozd"></Welcome>
    </div>
  );
}



