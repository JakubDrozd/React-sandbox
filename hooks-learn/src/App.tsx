import React, { useState } from 'react';
import logo from './logo.svg';
import './App.css';
import "bootstrap/dist/css/bootstrap.min.css"
import Button from "react-bootstrap/Button"

export default function App(props: any){
  const [count, setCount] = useState(0)

  const incrementCount = () =>{
    setCount(count + 1);
  }
  return (
    <div>
      <Button variant="info" onClick={incrementCount}>{count}</Button>
    </div>
  )
}