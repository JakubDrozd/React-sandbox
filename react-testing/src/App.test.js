import App from "./App";
import React from "react";
import { render, screen } from "@testing-library/react";
import { act } from "react-dom/test-utils";

it("should tick after one second to new value", () => {
  jest.useFakeTimers();
  render(<App></App>);
  const timer = screen.getByTestId("timer");
  expect(timer.innerHTML).toBe("0");
  act(() => {
    jest.advanceTimersByTime(1000);
  });
});
