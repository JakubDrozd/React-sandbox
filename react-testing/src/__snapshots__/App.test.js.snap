// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render magnificent monkeys 1`] = `
<div>
  <div>
    <button
      type="button"
    >
      Click me
    </button>
    <h1>
      Magnificent Monkeys
    </h1>
  </div>
</div>
`;
